### PR TITLE
increase read timeout to 5 mins due to long registration time

### LINKED ISSUE #13: Timeout during seed registration process
Nicholas ran into this problem in stage:

```
$ RAILS_ENV=production bundle exec rake was_registrar:register_crawls INPUT=/opt/app/was/was-registrar/shared/data/register_jobs_test_2016-06-23.tsv LOG='log/register_crawls.log'
Starting registering crawl task
/opt/app/was/was-registrar/shared/data/register_jobs_test_2016-06-23.tsv
Error in registering sul:SULWAS-DLSS-SINGLESITE-20140124213619  fom/20140124213614      druid:mc744qd2082       druid:md569kj5536
with #<RuntimeError: Error in registring the object. Request Timeout>
```

from the dor-services-app logs it looked like it worked from that side:

```
 INFO [2016-06-23 10:42:18] (10645)  :: #<Hashie::Mash admin_policy="druid:md569kj5536" collection="druid:mc744qd2082" initiate_work
flow="wasCrawlPreassemblyWF" label="fom/20140124213614" object_type="item" rights="dark" source_id="sul:SULWAS-DLSS-SINGLESITE-20140
124213619">
 INFO [2016-06-23 10:43:19] (10645)  :: {:admin_policy=>"druid:md569kj5536", :label=>"fom/20140124213614", :object_type=>"item", :ot
her_ids=>{:uuid=>"d8dfc880-3969-11e6-900d-0050569b52c6"}, :source_id=>{"sul"=>"SULWAS-DLSS-SINGLESITE-20140124213619"}, :tags=>[], :
initiate_workflow=>["wasCrawlPreassemblyWF"], :rights=>"dark", :collection=>"druid:mc744qd2082", :location=>"https://sul-dor-test.st
anford.edu/fedora/objects/druid:py765zw8342", :pid=>"druid:py765zw8342"}
171.67.23.28 - dorAdmin [23/Jun/2016:10:43:19 -0700] "POST /v1/objects HTTP/1.1" 201 17 63.9600
```

### PR DESCRIPTION
**Pull request by drh-stanford**
Fixes #13

### CODE REVIEW

**Comment by drh-stanford** [comment]
Marked WIP until I can finish testing this in stage.

Here's the backtrace for the timeout:

```
Error in registring the object. Request Timeout
Request Timeout: 
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/rest-client-1.8.0/lib/restclient/request.rb:427:in `rescue in transmit'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/rest-client-1.8.0/lib/restclient/request.rb:350:in `transmit'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/rest-client-1.8.0/lib/restclient/request.rb:176:in `execute'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/rest-client-1.8.0/lib/restclient/request.rb:41:in `execute'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/rest-client-1.8.0/lib/restclient.rb:69:in `post'
/opt/app/was/was-registrar/releases/20160624222228/lib/was/registrar/register_object.rb:32:in `register_object_using_web_service'
/opt/app/was/was-registrar/releases/20160624222228/lib/was/registrar/register_object.rb:17:in `register'
/opt/app/was/was-registrar/releases/20160624222228/app/controllers/seeds_controller.rb:56:in `register_one_item'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/actionpack-4.1.15/lib/action_controller/metal/implicit_render.rb:4:in `send_
action'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/actionpack-4.1.15/lib/abstract_controller/base.rb:189:in `process_action'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/actionpack-4.1.15/lib/action_controller/metal/rendering.rb:10:in `process_ac
tion'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/actionpack-4.1.15/lib/abstract_controller/callbacks.rb:20:in `block in proce
ss_action'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/activesupport-4.1.15/lib/active_support/callbacks.rb:113:in `call'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/activesupport-4.1.15/lib/active_support/callbacks.rb:113:in `call'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/activesupport-4.1.15/lib/active_support/callbacks.rb:552:in `block (2 levels) in compile'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/activesupport-4.1.15/lib/active_support/callbacks.rb:502:in `call'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/activesupport-4.1.15/lib/active_support/callbacks.rb:502:in `call'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/activesupport-4.1.15/lib/active_support/callbacks.rb:86:in `run_callbacks'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/actionpack-4.1.15/lib/abstract_controller/callbacks.rb:19:in `process_action'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/actionpack-4.1.15/lib/action_controller/metal/rescue.rb:29:in `process_action'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/actionpack-4.1.15/lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/activesupport-4.1.15/lib/active_support/notifications.rb:159:in `block in instrument'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/activesupport-4.1.15/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/activesupport-4.1.15/lib/active_support/notifications.rb:159:in `instrument'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/actionpack-4.1.15/lib/action_controller/metal/instrumentation.rb:30:in `process_action'
/opt/app/was/was-registrar/shared/bundle/ruby/2.1.0/gems/actionpack-4.1.15/lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
```

File: lib/was/registrar/register_object.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def register(item_hash)
       def register_object_using_web_service(register_params)
         Rails.logger.debug "Registering an object with params #{register_params}"
         begin
-          response = RestClient.post(Rails.configuration.service_root,  register_params, timeout: 60, open_timeout: 60)
+          response = RestClient.post(Rails.configuration.service_root,  register_params, timeout: 300, open_timeout: 60)
```
Comment (ndushay) [inline lib/was/registrar/register_object.rb]: this should be in a settings.yml file or something.

File: spec/registrar/register_object_spec.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
       registrar = Was::Registrar::RegisterObject.new
       allow(RestClient).to receive(:post).and_return(response)
-      expect(RestClient).to receive(:post).with('https://registr.stanford.edu/', params, timeout: 60, open_timeout: 60)
+      expect(RestClient).to receive(:post).with('https://registr.stanford.edu/', params, timeout: 300, open_timeout: 60)
```
Comment (ndushay) [inline spec/registrar/register_object_spec.rb]: these also in a settings.yml file.

**Comment by drh-stanford** [comment]
I was able to test this in stage and it works -- there's 2 timeouts -- one in the controller and another in the javascript ui.

I was successful in registering a seed, but it's still not generating the XML file that the `build-was-seed-druid-tree` robot needs. I will create a new issue for that.

See https://argo-stage.stanford.edu/catalog/druid:bs583pr6282

**Comment by ndushay** [comment]
Great progress!   Still would like to see timeout settings in a config file tho, not hardcoded.

**Comment by drh-stanford** [comment]
Re: `settings.yml`. We don't have that setup in `was-registrar` like we do in other rails apps. I can put it in the `config/application.rb` file as-is, or we can setup the `settings.yml` structure (which I don't know how to do offhand).

**Comment by ndushay** [comment]
tell ya what:  make a separate ticket for it and I'll merge this

**Comment by drh-stanford** [comment]
ok, added issue #24

**Pull request merged**